algebraic_topology/standard_topological_simplex.lean

```diff
@@ -0,0 +1,147 @@
+import algebraic_topology.simplex_category algebraic_topology.simplicial_set analysis.topology.topological_space analysis.nnreal
+noncomputable theory
+open finset
+
+local notation `ℝ≥0` := nnreal
+
+/- Powers of types -/
+section typepown
+
+instance typepown : has_pow Type* ℕ := ⟨λ α n, fin n → α⟩
+
+instance Rge0pown_topology {n : ℕ} : topological_space (ℝ≥0^n) :=
+by show topological_space (fin n → ℝ≥0); apply_instance
+
+end typepown
+
+/-- The n-th standard topological simplex: Δ_n = { x ∈ ℝ^{n+1} | ∀ i, x_i ≥ 0, and ∑ x_i = 1 } -/
+definition standard_topological_simplex (n : ℕ) : set (ℝ≥0^(n+1)) :=
```

Comment:
I added probability mass functions, you can use them instead of this construction.

Reply:
https://github.com/leanprover/mathlib/commit/4f9e951fbf403eb66d0814279bf929e37fec4fc0

Reply:
Aah, sorry, I didn't read "I added ..." as something you did today. My bad I didn't check the latest commits.

Reply:
@johoelzl I think I understand what those probability mass functions are doing. But I prefer to continue using `fin n`. Within this theory of simplicial sets etc, the simplex category plays quite an important role: you want to look at monotone functions `fin m \to fin n`. If I start using these probability mass functions, I fear losing access to those monotone functions. (Or at least their role in the theory it becomes a lot less apparent to me.)

Reply:
I don't understand?! You still use `fin n`: you would use `pmf (fin (n + 1))` to define `standard_topological_simplex n`. Then you use `pmf.map` to map your vertices.

Reply:
Ok, then I didn't understand `pmf` correctly. Let me see again.